src/main/webapp/app/exercises/text/assess/textblock-feedback-editor/textblock-feedback-editor.component.ts

```diff
@@ -142,4 +152,44 @@ export class TextblockFeedbackEditorComponent implements AfterViewInit {
         }
         this.didChange();
     }
+
+    // this method fires the modal service and shows a modal after connecting feedback with its respective blocks
+    async openOriginOfFeedbackModal(content: any) {
+        await this.connectAutomaticFeedbackOriginBlocksWithFeedback();
+        this.modalService.open(content, { size: 'lg' });
+    }
+
+    /**
+     * This method is used to find the submission used for making the current Automatic Feedback and retrieve its blocks.
+     * The blocks are then structured and set as a local property of this component to be shown in a modal
+     */
+    async connectAutomaticFeedbackOriginBlocksWithFeedback() {
+        // retrieve participation and submission references for the Automatic Feedback generated
+        const participationId = this.feedback.suggestedFeedbackParticipationReference ? this.feedback.suggestedFeedbackParticipationReference : -1;
+        const submissionId = this.feedback.suggestedFeedbackOriginSubmissionReference ? this.feedback.suggestedFeedbackOriginSubmissionReference : -1;
+        if (participationId >= 0 && submissionId >= 0) {
+            // finds the corresponding submission where the automatic feedback came from
+            const participation: StudentParticipation = await lastValueFrom(this.assessmentsService.getFeedbackDataForExerciseSubmission(participationId, submissionId));
+
+            // connect the feedback with its respective block if any.
+            let blocks: any[] = participation.submissions?.values().next().value.blocks;
+            // Sort blocks to show them in order.
+            blocks = blocks.sort((a, b) => a!.startIndex! - b!.startIndex!);
+            const feedbacks: any[] = participation.submissions?.values().next().value.latestResult.feedbacks;
+
+            // set list of blocks to be shown in the modal
+            this.listOfBlocksWithFeedback = blocks
+                .map((block) => {
+                    const blockFeedback = feedbacks.find((feedback) => feedback.reference === block.id);
+                    return {
+                        text: block.text,
+                        feedback: blockFeedback && blockFeedback.detailText,
+                        credits: blockFeedback ? blockFeedback.credits : 0,
+                        reusedCount: blockFeedback && block.numberOfAffectedSubmissions,
+                        type: this.feedback.suggestedFeedbackReference === block.id ? 'AUTOMATIC' : 'MANUAL',
+                    };
+                })
+                .filter((item) => item != undefined);
```

Comment:
Can we have undefined items here? It seems like we map every block into another structure that is not undefined.

Reply:
true, should have filtered by text instead. fixed. Thanks👍